### PR TITLE
Add geomesa feature list

### PR DESCRIPTION
**Pull request by moradology**
A (partial) list of geomesa features

### CODE REVIEW

File: GeoMesa_Feature_List.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    - Behavior tested; works
+  - Ingest vectors, provided a `GeoMesaInputFormat`
+    - Pointer: [IngestCommand.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/commands/IngestCommand.scala)
+    - Behavior tested; works
+  - Ingest rasters
+    - Pointer: [IngestRasterCommand.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/commands/IngestRasterCommand.scala)
+    - Supported file formats: "tif", "tiff", "geotiff", "dt0", "dt1", "dt2"
+
+
+- geomesa-convert (tools for converting various serialization formats to `SimpleFeature`s for ingestion - conversion mechanisms are specified by way of configuration files)
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: of note, we also have predefined converters for a variety of common data formats: https://github.com/locationtech/geomesa/tree/master/geomesa-tools/conf/sfts

File: GeoMesa_Feature_List.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+> The following document uses a hierarchical structure of features - from high level descriptions of general functionality to the subfeatures of which those general functions are composed.
+
+
+## Data Ingest/Input
+
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: we also support storm/kafka ingestion: http://www.geomesa.org/documentation/tutorials/geomesa-quickstart-storm.html

File: GeoMesa_Feature_List.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      - Apparently capable of reading from any GeoMesa `DataStore` as well as from the filesystem with or without avro files specifying the details of the conversion.
+- geomesa-process - (On Accumulo backed GeoMesa instances only - with the possible exception of `Point2Point` and `DensityProcess`, based on file locations and accumulo imports within said files. All processes are registered in https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/ProcessFactory.scala)
+  - computing a heatmap from a provided CQL query
+    - Pointer: [DensityProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/DensityProcess.scala)
+  - Given CQL and a description of the stats of interest, compute said stats on said CQL results
+    - Pointer: [StatsIteratorProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala)
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: stats we currently support: count, enumeration, frequency (countMinSketch), histogram, topk, min/max (bounds).
we also support count, bounds, histogram and top-k through the command line tools

Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: count, topk, min/max, histogram and frequency are also computed during ingestion and are available (without processing) for geometry, default date, and any attributes you indicate when creating your schema.

File: GeoMesa_Feature_List.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+
+## Output
+- geomesa-accumulo
+  - A reader for directly querying a datastore in java/scala
+    - Pointer: [<DataStore>.getFeatureReader](https://github.com/locationtech/geomesa/blob/99d2bcf47a2363f58e05abf7f3c39ef214551ed2/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L354)
+    - This is the best bet for high speed accumulo reads, per the GeoMesa gitter.
+  - Produce a collection of features for a given datastore
+    - Pointer: [<DataStore>.getFeatureSource](https://github.com/locationtech/geomesa/blob/99d2bcf47a2363f58e05abf7f3c39ef214551ed2/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L329)
+    - Performance characteristics vs the above reader are unclear. This feature is used, however, in the command line export
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: We also support direct output from map/reduce jobs: https://github.com/locationtech/geomesa/blob/master/geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaOutputFormat.scala

File: GeoMesa_Feature_List.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    - Pointer: [KNearestNeighborSearchProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/knn/KNearestNeighborSearchProcess.scala)
+  - Identify unique values for an attribute in results of a CQL query
+    - Pointer: [UniqueProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/unique/UniqueProcess.scala)
+  - Convert points to lines
+    - Pointer: [Point2PointProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/Point2PointProcess.scala)
+    - Convert a collection of points into a collection of line segments given a middle term parameter. Optionally break on the day of occurrence. This feature isn't really advertised.
+
+
+## Output
+- geomesa-accumulo
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: of note, all these outputs support relational projections

File: GeoMesa_Feature_List.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  - A reader for directly querying a datastore in java/scala
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: as more explanation - the feature reader/feature source/feature writer implement the OGC standard geotools framework.

File: GeoMesa_Feature_List.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  - Find the K nearest neighbors to a given point
+    - Pointer: [KNearestNeighborSearchProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/knn/KNearestNeighborSearchProcess.scala)
+  - Identify unique values for an attribute in results of a CQL query
+    - Pointer: [UniqueProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/unique/UniqueProcess.scala)
+  - Convert points to lines
+    - Pointer: [Point2PointProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/Point2PointProcess.scala)
+    - Convert a collection of points into a collection of line segments given a middle term parameter. Optionally break on the day of occurrence. This feature isn't really advertised.
+
+
+## Output
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: we also support output through geoserver using standard WFS/WMS queries

File: GeoMesa_Feature_List.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+- geomesa-stream (support for streaming input)
+  - A datastore which listens for updates from a [source which meets certain conditions](https://github.com/locationtech/geomesa/blob/7c295d68bad92291e4260273134219dd0f938faf/geomesa-stream/geomesa-stream-api/src/main/scala/org/locationtech/geomesa/stream/SimpleFeatureStreamSource.scala)
+    - Pointer: [StreamDataStore.scala](https://github.com/locationtech/geomesa/blob/master/geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala)
+    - A generic apache-camel based implementation](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-stream/geomesa-stream-generic/src/main/scala/org/locationtech/geomesa/stream/generic/GenericSimpleFeatureStreamSourceFactory.scala)
+  - Hooks for updating GeoServer on stream update
+    - Pointer: Implemented? [pomfile](https://github.com/locationtech/geomesa/tree/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-gs-plugin/geomesa-stream-gs-plugin)
+
+## Data Processing
+
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: another processing feature we support is data sampling - we haven't documented this well but an example can be seen here: https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/Z3IdxStrategyTest.scala#L301

File: GeoMesa_Feature_List.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  - Produce a collection of features for a given datastore
+    - Pointer: [<DataStore>.getFeatureSource](https://github.com/locationtech/geomesa/blob/99d2bcf47a2363f58e05abf7f3c39ef214551ed2/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L329)
+    - Performance characteristics vs the above reader are unclear. This feature is used, however, in the command line export
+- geomesa-tools (command line tools for interacting with geomesa)
+  - Serialize and export stored features (vectors)
+    - Pointer: [ExportCommand.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/commands/ExportCommand.scala)
+    - Supported export formats: CSV, shapefile, geojson, GML, BIN, Avro
+
+## Other Features
+
```
Comment (elahrvivaz) [inline GeoMesa_Feature_List.md]: other features we support:
* Google Bigtable backend
* multiple indices (including attribute indices) with a cost-based-optimizer for query planning (in accumulo)
* BLOB store - store large files in a pluggable way with geospatial query capabilities - http://www.geomesa.org/documentation/tutorials/geomesa-blobstore.html
* Native API - store arbitrary java objects with geospatial query capabilities - https://github.com/locationtech/geomesa/tree/master/geomesa-native-api

**Comment by moradology** [comment]
I made sure to include all the changes mentioned here - if we come across anything else, I'll try to update this as we move forward.

**Pull request merged**